manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -373,7 +373,13 @@ data:
       {{- if .Values.global.proxy.lifecycle }}
         lifecycle:
           {{ toYaml .Values.global.proxy.lifecycle | indent 4 }}
-        {{- end }}
+      {{- else if eq (annotation .ObjectMeta `alpha.sidecar.istio.io/holdApplicationUntilProxyStarts` .Values.global.proxy.holdApplicationUntilProxyStarts) `true`}}
+        lifecycle:
+          postStart:
+            exec:
+              command:
+              - /usr/local/bin/wait-ready.sh
```

Comment:
Will probably fail in 'distroless', I don't think we have a shell.

Reply:
In other words - 'istio-agent wait' instead of the shell.

Reply:
Working on implementing `istio-agent wait` right now.